src/main/java/org/jenkinsci/plugins/durabletask/WindowsBatchScript.java

```diff
@@ -59,34 +86,79 @@ public String getScript() {
         if (launcher.isUnix()) {
             throw new IOException("Batch scripts can only be run on Windows nodes");
         }
+
         BatchController c = new BatchController(ws);
 
-        String cmd;
+        List<String> launcherCmd = null;
+        FilePath binary;
+        if (FORCE_BINARY_WRAPPER && (binary = requestBinary(ws, c)) != null) {
+            launcherCmd = binaryLauncherCmd(c, ws, binary.getRemote(), c.getBatchFile2(ws).getRemote());
+            c.getBatchFile2(ws).write(script, "UTF-8");
+        }
+        if (launcherCmd == null) {
+            launcherCmd = scriptLauncherCmd(c, ws);
+        }
+
+        LOGGER.log(Level.FINE, "launching {0}", launcherCmd);
+        Launcher.ProcStarter ps = launcher.launch().cmds(launcherCmd).envs(escape(envVars)).pwd(ws).quiet(true);
+
+        /* Too noisy, and consumes a thread:
+        ps.stdout(listener);
+        */
+        ps.readStdout().readStderr(); // TODO see BourneShellScript
+        Proc p = ps.start();
+        c.registerForCleanup(p.getStdout());
+        c.registerForCleanup(p.getStderr());
+
+        return c;
+    }
+
+    @Nonnull
+    private List<String> binaryLauncherCmd(BatchController c, FilePath ws, String binaryPath, String scriptPath) throws IOException, InterruptedException {
+        String logFile = c.getLogFile(ws).getRemote();
+        String resultFile = c.getResultFile(ws).getRemote();
+        String outputFile = c.getOutputFile(ws).getRemote();
+        String controlDirPath = c.controlDir(ws).getRemote();
+
+        List<String> cmd = new ArrayList<>();
+        cmd.add(binaryPath);
+        cmd.add("-daemon");
+        cmd.add("-executable=cmd");
+        cmd.add("-args=/C call \\\"" + scriptPath + "\\\"");
+        cmd.add("-controldir=" + controlDirPath);
+        cmd.add("-result=" + resultFile);
+        cmd.add("-log=" + logFile);
+        if (capturingOutput) {
+            cmd.add("-output=" + outputFile);
+        }
+        if (LAUNCH_DIAGNOSTICS) {
+            cmd.add("-debug");
+        }
+        return cmd;
+    }
+
+    @Nonnull
+    private List<String> scriptLauncherCmd(BatchController c, FilePath ws) throws IOException, InterruptedException {
+
+        String cmdString;
         if (capturingOutput) {
-            cmd = String.format("@echo off \r\ncmd /c call \"%s\" > \"%s\" 2> \"%s\"\r\necho %%ERRORLEVEL%% > \"%s\"\r\n",
+            cmdString = String.format("@echo off \r%ncmd /c call \"%s\" > \"%s\" 2> \"%s\"\r%necho %%ERRORLEVEL%% > \"%s\"\r%n",
```

Comment:
Not convinced here with the String format.
`%n` may be `\r\n` or `\n` depending on if the controller is running on Windows or not.

We really do not want this to be platform dependant new lines - we really do want to force it to be windows as this is batch script and they like windows line ends.  (not sure how this would cope with `\r\r\n` - it may work - it generally does in later version of windows with just `\n` but the output here would be funky and possibly harder to debug than it should be).

```suggestion
            cmdString = String.format("@echo off \r\ncmd /c call \"%s\" > \"%s\" 2> \"%s\"\r\necho %%ERRORLEVEL%% > \"%s\"\r\n",
```

Reply:
I'm perfectly fine with keeping it `\n`. This was only changed because spotbugs reported as an error. I would just need to add a `SUPPRESS` I believe.

Reply:
hmm, interesting, i think it was just my windows machine that was failing with that spotbug error. There's clearly a suppress for this particular issue.